lib/rubygems/util/spdx.rb

```diff
@@ -0,0 +1,303 @@
+      Cube
+      D-FSL-1.0
+      diffmark
+      DOC
+      Dotseqn
+      DSDP
+      dvipdfm
+      ECL-1.0
+      ECL-2.0
+      EFL-1.0
+      EFL-2.0
+      eGenix
+      Entessa
+      EPL-1.0
+      ErlPL-1.1
+      EUDatagrid
+      EUPL-1.0
+      EUPL-1.1
+      Eurosym
+      Fair
+      Frameworx-1.0
+      FreeImage
+      FSFUL
+      FSFULLR
+      FTL
+      GFDL-1.1
+      GFDL-1.2
+      GFDL-1.3
+      Giftware
+      GL2PS
+      Glide
+      Glulxe
+      gnuplot
+      GPL-1.0
+      GPL-2.0
+      GPL-3.0
+      gSOAP-1.3b
+      HaskellReport
+      HPND
+      IBM-pibs
+      ICU
+      IJG
+      ImageMagick
+      iMatix
+      Imlib2
+      Intel
+      Intel-ACPI
+      IPA
+      IPL-1.0
+      ISC
+      JasPer-2.0
+      JSON
+      Latex2e
+      Leptonica
+      LGPL-2.0
+      LGPL-2.1
+      LGPL-3.0
+      Libpng
+      libtiff
+      LPL-1.0
+      LPL-1.02
+      LPPL-1.0
+      LPPL-1.1
+      LPPL-1.2
+      LPPL-1.3a
+      LPPL-1.3c
+      MakeIndex
+      MirOS
+      MIT
+      MIT-advertising
+      MIT-CMU
+      MIT-enna
+      MIT-feh
+      MITNFA
+      Motosoto
+      mpich2
+      MPL-1.0
+      MPL-1.1
+      MPL-2.0
+      MPL-2.0-no-copyleft-exception
+      MS-PL
+      MS-RL
+      MTLL
+      Multics
+      Mup
+      NASA-1.3
+      Naumen
+      NBPL-1.0
+      NCSA
+      NetCDF
+      Newsletr
+      NGPL
+      NLPL
+      Nokia
+      NOSL
+      Noweb
+      NPL-1.0
+      NPL-1.1
+      NPOSL-3.0
+      NRL
+      NTP
+      Nunit
+      OCLC-2.0
+      ODbL-1.0
+      OFL-1.0
+      OFL-1.1
+      OGTSL
+      OLDAP-1.1
+      OLDAP-1.2
+      OLDAP-1.3
+      OLDAP-1.4
+      OLDAP-2.0
+      OLDAP-2.0.1
+      OLDAP-2.1
+      OLDAP-2.2
+      OLDAP-2.2.1
+      OLDAP-2.2.2
+      OLDAP-2.3
+      OLDAP-2.4
+      OLDAP-2.5
+      OLDAP-2.6
+      OLDAP-2.7
+      OLDAP-2.8
+      OML
+      OpenSSL
+      OPL-1.0
+      OSL-1.0
+      OSL-1.1
+      OSL-2.0
+      OSL-2.1
+      OSL-3.0
+      PDDL-1.0
+      PHP-3.0
+      PHP-3.01
+      Plexus
+      PostgreSQL
+      psfrag
+      psutils
+      Python-2.0
+      Qhull
+      QPL-1.0
+      Rdisc
+      RHeCos-1.1
+      RPL-1.1
+      RPL-1.5
+      RPSL-1.0
+      RSA-MD
+      RSCPL
+      Ruby
+      Saxpath
+      SAX-PD
+      SCEA
+      SGI-B-1.0
+      SGI-B-1.1
+      SGI-B-2.0
+      SimPL-2.0
+      SISSL
+      SISSL-1.2
+      Sleepycat
+      SMLNJ
+      SNIA
+      SPL-1.0
+      SugarCRM-1.1.3
+      SWL
+      TCL
+      TMate
+      TORQUE-1.1
+      TOSL
+      Unicode-TOU
+      Unlicense
+      UPL-1.0
+      Vim
+      VOSTROM
+      VSL-1.0
+      W3C
+      W3C-19980720
+      Watcom-1.0
+      Wsuipa
+      WTFPL
+      X11
+      Xerox
+      XFree86-1.1
+      xinetd
+      Xnet
+      xpp
+      XSkat
+      YPL-1.0
+      YPL-1.1
+      Zed
+      Zend-2.0
+      Zimbra-1.3
+      Zimbra-1.4
+      Zlib
+      zlib-acknowledgement
+      ZPL-1.1
+      ZPL-2.0
+      ZPL-2.1
+    ).freeze
```

Comment:
@bf4, I brought the list in manually.

Others have taken the scraping route, like @sindresorhus in the Node world with [`spdx-license-list`](https://github.com/sindresorhus/spdx-license-list). That module requires a manual step between running the scraping routine and committing a list to a repository for public consumption, and brings in external dependencies to parse an `xls`, the most direct expression of SPDX working group truth. Any breakage in the scraping routine due to file or format changes can be caught prepublication.

I suppose the `generate_tables` approach would mean that someday, some poor sap's otherwise green commit will fail red because SPDX.org changed. Whether that eventual pain is worth the marginal improvement to license list "correctness" in the meantime strikes me as a trade-off for maintainers to assess. More than landing this pull request, I want respect the incredible amount of time and discipline that go into RubyGems.

For what it's worth, the better part of "correctness" can be had with `MIT`, `Apache-2.0`, the `BSD-` identifiers, and the various `GPL-` versions alone. A scraper would serve mostly to hush warnings about licenses that very, very few RubyGem authors choose. In putting together this PR, frankly, I thought more about trimming weight by omitting rare license IDs than scraping an up-to-date list from the canonical source. If I drew the line too narrow or broad, I'll be very happy to amend.